include/glow/Backends/CompiledFunction.h

```diff
@@ -16,23 +16,50 @@
 #ifndef GLOW_BACKENDS_COMPILEDFUNCTION_H
 #define GLOW_BACKENDS_COMPILEDFUNCTION_H
 
+#include "glow/Graph/Nodes.h"
 #include <unordered_map>
 
 namespace glow {
 
 class Context;
-
+namespace runtime {
+/// RuntimeSymbolInfo
+/// Contains information for initialization and handling of symbol at runtime.
+struct RuntimeSymbolInfo {
+  /// The size in bytes.
+  size_t size;
+  /// Offset in bytes from the base address.
+  size_t offset;
+};
+/// Runtime Bundle
+/// Contains the information needed to be passed forward from compile time to
+/// runtime. In order to allocate and initialize memory.
+struct RuntimeBundle {
+  /// Map from symbol name to a RuntimeSymbolObject.
+  std::unordered_map<std::string, RuntimeSymbolInfo> symbolTable;
+  /// Pointer to memory containing the weights for execution.
+  uint8_t *constants;
+  /// Amount of memory needed for weights.
+  const size_t constantWeightVarsMemSize;
+  /// Mmount of memory needed for mutable vars.
+  const size_t mutableWeightVarsMemSize;
+  /// Amount of memory needed for activations.
+  const size_t activationsMemSize;
+  RuntimeBundle(size_t constWeight, size_t mutableWeight, size_t activations)
+      : constantWeightVarsMemSize(constWeight),
+        mutableWeightVarsMemSize(mutableWeight),
+        activationsMemSize(activations) {}
```

Comment:
It feels weird that this API doesn't have anything to manage the symbolTable and the constants array.
It feels weird as well to have the size for the various section but the pointer only for the constants. Could you elaborate on how to you see this being used?

Reply:
These fields could be passed to the CompiledFunction directly but a struct gives us flexibility going forward to support different backend needs.

Reply:
What I am trying to get at is from a design perspective, one could mess with the constants pointer without doing the proper changes to the weight size member. That's why I am surprised it is not encapsulate behind some APIs. I.e., we expect the users of this API to do the right thing and that's when I get suspicious :).

Now, from a more high level point of view, the handling of constants and mutable allocations are the same. The only difference is that we know the contain of the constants at compile time. Therefore what I am questioning is whether or not this special handling of constants makes sense.